### PR TITLE
refactor!: refactor how we do defaults for runtime overrides

### PR DESCRIPTION
**Pull request by slim-bean**
**What this PR does / why we need it**:

We introduced a "defaults" section into the runtime overrides for runtime configs, however this prevented us from being able to set a default for these values without providing a runtime config.

we already have a pattern for this we used for limits_config so this PR refactors those defaults to be a first class object in the loki config and follows the same pattern of setting this as defaults.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/loki/loki.go [hunk 1]
```diff
@@ -99,9 +99,10 @@ type Config struct {
 	TableManager        index.TableManagerConfig   `yaml:"table_manager,omitempty"`
 	MemberlistKV        memberlist.KVConfig        `yaml:"memberlist"`
 
-	RuntimeConfig runtimeconfig.Config `yaml:"runtime_config,omitempty"`
-	Tracing       tracing.Config       `yaml:"tracing"`
-	Analytics     analytics.Config     `yaml:"analytics"`
+	RuntimeConfig   runtimeconfig.Config `yaml:"runtime_config,omitempty"`
+	OperationConfig runtime.Config       `yaml:"operation_config,omitempty"`
```
Comment (slim-bean) [inline pkg/loki/loki.go:103]: struggling a bit with the name here.....

File: pkg/runtime/config.go [hunk 2]
```diff
@@ -9,7 +11,26 @@ type Config struct {
 	LimitedLogPushErrors bool `yaml:"limited_log_push_errors"`
 }
 
-var EmptyConfig = &Config{}
+// RegisterFlags adds the flags required to config this to the given FlagSet
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.BoolVar(&cfg.LogStreamCreation, "operation-config.log-stream-creation", false, "Log every new stream created by a push request (very verbose, recommend to enable via runtime config only).")
+	f.BoolVar(&cfg.LogPushRequest, "operation-config.log-push-request", false, "Log every push request (very verbose, recommend to enable via runtime config only).")
+	f.BoolVar(&cfg.LogPushRequestStreams, "operation-config.log-push-request-streams", false, "Log every stream in a pus request (very verbose, recommend to enable via runtime config only).")
```
Comment (DylanGuedes) [inline pkg/runtime/config.go:18]:
```suggestion
	f.BoolVar(&cfg.LogPushRequestStreams, "operation-config.log-push-request-streams", false, "Log every stream in a push request (very verbose, recommend to enable via runtime config only).")
```

**Review by DylanGuedes — CHANGES_REQUESTED** [review]
looks awesome.

could you add a changelog entry (since it is a breaking change)?

**Comment by slim-bean** [comment]
> could you add a changelog entry (since it is a breaking change)?

it's only a breaking change for us right now, I don't believe we ever released code with that default section in the runtime config? that didn't make it into 2.9 I don't think?

File: production/ksonnet/loki/overrides.libsonnet [hunk 3]
```diff
@@ -25,14 +25,6 @@ local k = import 'ksonnet-util/kausal.libsonnet';
...
-
-    defaultRuntimeConfigs: {
-      // override the default runtime configs. Useful to change a behavior globally across multiple tenants. See pkg/loki/runtime_config.go
-      //
-      // log_stream_creation: true,
-      // log_push_request: true,
-      // log_push_request_streams: true,
-    },
   },
   local configMap = k.core.v1.configMap,
```
Comment (DylanGuedes) [inline production/ksonnet/loki/overrides.libsonnet:30]: i think you'll have to remove this line too
```diff
  overrides_config:
    configMap.new($._config.overrides_configmap_name) +
    configMap.withData({
      'overrides.yaml': k.util.manifestYaml(
        {
          overrides: $._config.overrides,
          configs: $._config.runtimeConfigs,
-         default: $._config.defaultRuntimeConfigs,
        }
        + (if std.length($._config.multi_kv_config) > 0 then { multi_kv_config: $._config.multi_kv_config } else {}),
      ),
    }),
```

**Review by DylanGuedes — CHANGES_REQUESTED** [review]

File: production/ksonnet/loki/overrides.libsonnet (hunk 3 above)
Comment (slim-bean) [reply production/ksonnet/loki/overrides.libsonnet:30]: ah good catch, ty

**Review by DylanGuedes — APPROVED** [review]

**Pull request merged**